src/ctapipe_io_lst/__init__.py

```diff
@@ -686,14 +686,12 @@ def fill_trigger_info(self, array_event):
             )
 
         trigger.event_type = self._event_type_from_trigger_bits(trigger_bits)
-        
+
         if trigger.event_type == EventType.FLATFIELD:
-            waveform = array_event.r0.tel[tel_id].waveform
-            if waveform is None:
-                waveform = array_event.r1.tel[tel_id].waveform
-            if waveform.ndim == 2:
+            waveform = array_event.r1.tel[tel_id].waveform
+            if waveform is not None and waveform.ndim == 2:
```

Comment:
This is not identical to the previous code - just to make sure: we never need to check the dimensions of the R0 waveform? Is it always 3, in case R0 (and not R1) is filled?

Reply:
Anyway something is wrong with my earlier changes - test of lstchain_data_r0_to_dl1 on 12714.0100 does not work with --flatfield-heuristic (and it does work with the main branch)

Reply:
Ah, the problem was this:

https://github.com/cta-observatory/ctapipe_io_lst/blob/d0e5845f20fd32885ff54de333d806967f92e2dd/src/ctapipe_io_lst/__init__.py#L715-L716

Because now the event type of events trigger-tagged FF and not looking like FF is not set to UNKNOWN.

Reply:
> This is not identical to the previous code - just to make sure: we never need to check the dimensions of the R0 waveform? Is it always 3, in case R0 (and not R1) is filled?

Yes, R0 is always two-gain (for LST)

Reply:
Only the R1 container has the `selected_gain` field, this is why we put single-gain data into the R1 container even before the calibration...